tianshou/highlevel/module/module_opt.py

```diff
@@ -0,0 +1,29 @@
+from dataclasses import dataclass
+
+import torch
+
+from tianshou.utils.net.common import ActorCritic
+
+
+@dataclass
+class ModuleOpt:
+    """Container for a torch module along with its optimizer."""
+
+    module: torch.nn.Module
+    optim: torch.optim.Optimizer
+
+
+@dataclass
+class ActorCriticModuleOpt:
```

Comment:
Which typing issue are you referring to? Mypy reports no issues on my end.

An `ActorCriticModuleOpt` instance can never be used in place of a `ModuleOpt`. No Union typing is being used, the two classes are used in distinct cases. I realize that the name of the former class perhaps suggests a type relationship, but there isn't one. I think the best solution is to rename `ActorCriticModuleOpt` to `ActorCriticOpt` in order to avoid this confusion.